src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/runInTerminalTool.ts

```diff
@@ -408,7 +408,7 @@ export class RunInTerminalTool extends Disposable implements IToolImpl {
 		const cdPrefixMatch = commandLine.match(
 			isPwsh
 				? /^(?:cd|Set-Location(?: -Path)?) (?<dir>[^\s]+) ?(?:&&|;)\s+(?<suffix>.+)$/i
-				: /^cd (?<dir>[^\s]+) &&\s+(?<suffix>.+)$/
+				: /^cd(?: \/d)? (?<dir>(?:[^\s]+|"[^"]*")) ?(?:&&|;)\s*(?<suffix>.+)$/
```

Comment:
The /d is a Windows thing, so it should live in the isPwsh branch. Don't touch this line